src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -29,19 +36,28 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
     val executionContext: ExecutionContext,
     val openTelemetry: OpenTelemetryMetrics[IO]
 ) extends LazyLogging {
-
   def createUser(possibleNewUser: SamUser, samRequestContext: SamRequestContext): IO[UserStatus] =
+    createUser(possibleNewUser, None, samRequestContext)
+
+  def createUser(possibleNewUser: SamUser, registrationRequest: Option[SamUserRegistrationRequest], samRequestContext: SamRequestContext): IO[UserStatus] =
     openTelemetry.time("api.v1.user.create.time", API_TIMING_DURATION_BUCKET) {
       // Validate the values set on the possible new user, short circuit if there's a problem
       val validationErrors = validateUser(possibleNewUser)
       if (validationErrors.nonEmpty) {
         return IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "invalid user", validationErrors.get)))
       }
+      val registrationRequestErrors = registrationRequest.flatMap(_.validateForNewUser)
+      if (registrationRequestErrors.nonEmpty) {
+        return IO.raiseError(
+          new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "invalid registration request", registrationRequestErrors.get))
```

Comment:
does the ui pr have to go out first for this, so that it has the required fields sent with the registration request?

Reply:
This is tied into a new endpoint which isn't being hit yet. If the request object doesn't exist, like it currently doesn't, these errors won't exist.

Reply:
Right ok and if the consent boolean isnt there itll default to false is what im reading?

Reply:
yup!